hepstats/hypotests/toyutils.py

```diff
@@ -0,0 +1,356 @@
+import asdf
+import os
+import numpy as np
+
+from .parameters import POI, POIarray
+from .exceptions import ParameterNotFound
+from .fitutils.utils import pll
+from .fitutils.sampling import base_sampler, base_sample
+from .hypotests_object import ToysObject
+
+"""
+Module defining the classes to perform and store the results of toy experiments.
+
+Acronyms used in the code:
+    * nll = negative log-likehood, the likehood being the `loss` attribute of a calculator;
+"""
+
+
+class ToyResult(object):
+
+    def __init__(self, poigen, poieval):
+        """
+        Class to store the results of toys generated for a given value of a POI.
+        The best fit value of the POI, the NLL evaluate at the best fit, and the NLL evaluated
+        at several values of the POI are stored.
+
+        Args:
+            poigen (POI): POI used to generate the toys
+            poieval (POIarray): POI values to evaluate the loss function
```

Comment:
It could be useful to mention the way to get the actual result, which is only implemented via a dict for now. And it would make sense to give the key list of the returned result in the `to_dict` method, as that's relevant info for the user.